### ISSUE TITLE
Where Filtering 

### ISSUE THREAD

**Issue by DregondRahl** [maoosi/prisma-appsync#9]
Where filtering for custom resolvers doesn't handle the args.where, would need to split/convert (parseWhere) to {name: { contains: "rai" } }

```
{
    "operation": "custom",
    "model": "paginateUsers",
    "args": {
        "select": {
            "name": true,
            "id": true
        },
        "where": {
            "name_contains": "rai"
        },
        "skip": 0,
        "take": 10
    }
}
```

Alternatively, if the where Inputs types that are generated  were converted to follow a similar structure it would become more closer to how prisma and nexus handles filtering.

```
extend type Query {
  paginateUsers(
    where: PaginateUserWhereInput
    orderBy: UserOrderByInput
    skip: Int
    take: Int
  ): [User]
}

input PaginateUserWhereInput {
  AND: [UserWhereInput!]
  OR: [UserWhereInput!]
  NOT: [UserWhereInput!]
  id: Int
  name: StringNullableFilter
}

input StringNullableFilter {
  equals: String
  in: [String!]
  notIn: [String!]
  lt: String
  lte: String
  gt: String
  gte: String
  contains: String
  startsWith: String
  endsWith: String
  not: NestedStringNullableFilter
}

input NestedStringNullableFilter {
  equals: String
  in: [String!]
  notIn: [String!]
  lt: String
  lte: String
  gt: String
  gte: String
  contains: String
  startsWith: String
  endsWith: String
  not: NestedStringNullableFilter
}
```

![image](https://user-images.githubusercontent.com/749080/115106469-b53a6500-9f82-11eb-9786-04c2872705a2.png)

**Comment by maoosi** [comment]
Thanks for sharing on this @DregondRahl !

Custom resolvers are intentionally excluded from the automatic parsing/transformations done on all the CRUD operations. Though it would be possible to expose some of the internal utility functions as part of the `CustomResolverProps` to help with such a scenario.

Now I like your alternative suggestion better, as it would help simplify both usage and implementation. I’ll do some experiment with this as soon as I have some time and keep you posted.

Pagination is already part of the roadmap https://github.com/maoosi/prisma-appsync/issues/6

**Comment by maoosi** [comment]
Pagination and new filtering system available in [1.0.0-beta.56](https://github.com/maoosi/prisma-appsync/releases/tag/1.0.0-beta.56).
